index.bs

```diff
@@ -12089,6 +12114,102 @@ running the algorithm for an {{AudioNode}}, using an <a>input
 buffer</a> and the value(s) of the {{AudioParam}}(s) of this
 {{AudioNode}} as the input for this algorithm.
 
+<h3 id="interruption-handling">Handling an interruption on the {{AudioContext}}</h3>
+
+An <dfn>interruption</dfn> is an event generated by the user agent when it needs
+to halt audio playback for an {{AudioContext}}. For example, The user agent may
+create an interruption when another application requests exclusive access to the
+audio output hardware.
+
+When an [=interruption=] happens, the user agent must <a>queue a control message</a>
+to interrupt the {{AudioContext}}.
+
+Running a <a>control message</a> to <dfn attribute for="AudioContext"
+lt="interruption-start">interrupt an {{AudioContext}}</dfn> |context|
+means running these steps on the <a>rendering thread</a>:
+
+1. If the |context|'s {{[[rendering thread state]]}} is <code>closed</code>
+    or <code>interrupted</code>:
+
+    1. Abort these steps.
+
+1. If the |context|'s {{[[rendering thread state]]}} is
+    <code>running</code>:
+
+    1. Attempt to <a>release system resources</a>.
+
+    1. [=Queue a media element task=] to execute the following steps:
+
+        1. Set the |context|'s {{[[control thread state]]}} to
+            <code>interrupted</code>.
+
+        1. [=Fire an event=] named {{BaseAudioContext/statechange}} at the
+            |context|.
+
+1. If the |context|'s {{[[rendering thread state]]}} is
+    <code>suspended</code>:
+
+    1. [=Queue a media element task=] to execute the following steps:
+
+        1. Set the |context|'s {{[[control thread state]]}} to
+            <code>interrupted</code>.
+
+1. Set the |context|'s {{[[rendering thread state]]}} to
+    <code>interrupted</code>.
+
+Note: If the {{AudioContext}} is <code>suspended</code> a
+{{BaseAudioContext/statechange}} event is not fired for privacy reasons to avoid
+over-sharing user activity - e.g. when a phone call comes in or when the screen
+gets locked.
+
+When an [=interruption=] ends, the user agent must <a>queue a control message</a>
+to <dfn attribute for="AudioContext" lt="interruption-end">end the
+{{AudioContext}} interruption</dfn>.
+
+<dfn lt="linear PCM">Linear pulse code modulation</dfn>
+
+Running a <a>control message</a> to end an {{AudioContext}} |context|
+[=interruption=] means running these steps on the <a>rendering thread</a>:
+
+1. If the |context|'s {{[[rendering thread state]]}} is not
+    <code>interrupted</code>:
+
+    1. Abort these steps.
+
+1. If the |context|'s {{[[rendering thread state]]}} was
+    <code>running</code> before the [=interruption=] or if the the
+    |context|'s {{[[rendering thread state]]}} was <code>suspended</code>
+    and {{AudioContext/resume}} was called during the [=interruption=]:
```

Comment:
We want a state for this. Store the "previous state" in a private slot and reference it. This way, it's clear what happens, we can describe it algorithmically.

Reply:
Sounds good. Thanks! I wrote the algorithms like this because I remember that we wanted to be more handwavy with respect to how these states are managed by the UA. But I like to be able to have the clear algorithm steps. 

To address that, I created a new `BaseAudioContext` private slot called `[[interrupt end return state]]`. I don't know if this is the best name, but I couldn't think of a better one. I also rewrote the algorithm for the "interruption end" to branch based on the value of `[[interrupt end return state]]`.